tests/pytest/eligibility/test_views.py

```diff
@@ -1,204 +1,272 @@
+    mock_session = mocker.patch("benefits.eligibility.views.session")
+    mock_session.logged_in.return_value = False
 
-    with file_path.open(mode="rb") as pemfile:
-        key = str(pemfile.read(), "utf-8")
+    path = reverse(ROUTE_START)
+    response = client.get(path)
 
-    return key
+    assert response.status_code == 200
+    assert "page" in response.context_data
+    assert len(response.context_data["page"].buttons) == 1
+    assert response.context_data["page"].buttons[0].url == reverse(ROUTE_LOGIN)
 
 
-@httpretty.activate(verbose=True, allow_net_connect=False)
 @pytest.mark.django_db
-@pytest.mark.parametrize(
-    "exception", [requests.ConnectionError, requests.Timeout, requests.TooManyRedirects, requests.HTTPError]
-)
-def test_confirm_failure_error_on_request(mocker, rf, exception):
-    agency, verifier = set_verifier(mocker)
+@pytest.mark.usefixtures("mocked_session_agency", "mocked_verifier_form", "mocked_session_verifier_auth_not_required")
+def test_start_verifier_auth_not_required(client):
+    path = reverse(ROUTE_START)
+    response = client.get(path)
 
-    def raise_exception(*args, **kwargs):
-        raise exception()
+    assert response.status_code == 200
+    assert "page" in response.context_data
+    assert len(response.context_data["page"].buttons) == 1
+    assert response.context_data["page"].buttons[0].url == reverse(ROUTE_CONFIRM)
 
-    mocker.patch("requests.get", new=raise_exception)
 
-    path = reverse("eligibility:confirm")
-    body = {"sub": "A7654321", "name": "Garcia"}
-    request = rf.post(path, body)
+@pytest.mark.django_db
+@pytest.mark.usefixtures("mocked_session_agency")
+def test_start_without_verifier(client):
+    path = reverse(ROUTE_START)
+    with pytest.raises(AttributeError, match=r"verifier"):
+        client.get(path)
 
-    initialize_request(request)
-    session.update(request, agency=agency, verifier=verifier, oauth_token="token")
 
-    with pytest.raises(ApiError):
-        confirm(request)
+@pytest.mark.django_db
+@pytest.mark.usefixtures("mocked_eligibility_auth_request")
+def test_confirm_get_unverified(client):
+    path = reverse(ROUTE_CONFIRM)
+    response = client.get(path)
+
+    assert response.status_code == 200
+    assert response.template_name == TEMPLATE_CONFIRM
 
 
-@httpretty.activate(verbose=True, allow_net_connect=False)
 @pytest.mark.django_db
-def test_confirm_failure_unexpected_status_code(mocker, rf):
-    agency, verifier = set_verifier(mocker)
+@pytest.mark.usefixtures("mocked_eligibility_auth_request", "mocked_session_eligibility")
+def test_confirm_get_verified(client, mocked_session_update):
+    path = reverse(ROUTE_CONFIRM)
+    response = client.get(path)
 
-    httpretty.register_uri(httpretty.GET, "http://localhost/verify", status=404)
+    assert response.status_code == 302
+    assert response.url == reverse(ROUTE_ENROLLMENT)
+    mocked_session_update.assert_called_once()
```

Comment:
Some tests in `eligibility/test_views.py` are failing for me because they assert that `session.update` is called once, but if you have DEBUG mode set to True, the `DebugSession` middleware will call `session.update`, causing the test to fail.

![image](https://user-images.githubusercontent.com/25497886/173460349-09e2e609-d4d8-485c-89c6-89e003476ae1.png)

![image](https://user-images.githubusercontent.com/25497886/173463058-cc1995cb-22d2-4d84-9c4e-b3328347ab3d.png)

![image](https://user-images.githubusercontent.com/25497886/173463072-de7522a7-416f-4ab9-a9f3-7d0e530b32ec.png)

Should we disable the `DebugSession` middleware when running these tests?

Reply:
Interesting! I had not run the tests with `DJANGO_DEBUG=true` sorry about that! Will look into this.

Reply:
@angela-tran do you have `DJANGO_DEBUG=True` in your `.env` file?

Reply:
@thekaveman Yes - it's `DJANGO_DEBUG=true`. When I change it to `DJANGO_DEBUG=false` and rebuild the devcontainer, re-run the tests, all the tests pass.

Reply:
OK, can we take this as a follow-up then?

With #598 we added that env override to `launch.json`, so the app always launches in debug mode from the devcontainer `F5`, regardless of the `.env` file variable. You don't _actually_ need this variable in your local `.env` anymore.

Here it looks like the `.env` file variable is impacting the test runs, and I think we could go either way in supporting the tests under debug mode and not? Since we shouldn't be running the app in debug mode anywhere out there.

Reply:
Thanks for reminding me that `launch.json` can help us easily develop in with DEBUG mode on AND run the tests locally with DEBUG mode off. I'd say we don't even need a followup if we want to go ahead and say the tests don't support DEBUG mode.